references/cli/provider.go

```diff
@@ -94,7 +102,11 @@ func NewProviderListCommand(c common.Args, ioStreams cmdutil.IOStreams) *cobra.C
 }
 
 func prepareProviderAddCommand(c common.Args, ioStreams cmdutil.IOStreams) (*cobra.Command, error) {
-	ctx := context.Background()
+	if len(os.Args) < 2 || os.Args[1] != "provider" {
+		return nil, nil
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute*1)
```

Comment:
What are the advantages of lazy-loading the sub-commands `vela provider add` and `vela provider delete`? To help speed up the loading of other commands?